packages/scene-composer/src/interfaces/components.ts

```diff
@@ -49,6 +50,7 @@ export interface IAnchorComponent extends ISceneComponent {
   offset?: Vector3;
   chosenColor?: string;
   customColors?: string[];
+  customIcon?: IIconDefinition;
```

Comment:
the `IIconDefinition` has too much info to be saved into scene file. do we really need to support customized icon height width? I thought we will just need the icon name/id so we know which icon to render, then our code will handle sizing stuff

Reply:
No. I also thought `faInfo` is a name but its an object of type IconDefinition. Icondefinition has prefix, name and icon. Their api accepts an icon of type Icondefinition. And each IconDefinition has different configuration that means different height/width every time user selects on the page. Icon prop has an of configuration of that icon including path to svg. We really can't skip any of this as it is same as saving databinding object. I don't think this will introduce any issues.

Reply:
the icon has type 
```
export type IconProp = IconName | [IconPrefix, IconName] | IconLookup;
export interface IconLookup {
  prefix: IconPrefix;
  iconName: IconName;
}
```
so prefix + name is good enough

Reply:
In this approach, I will have to internally search every time for this IconName and get the IconDefinition. Because I don't think IconName give us theIconDefinition out of the box.